### PR TITLE
Implement Rex::Text random function name generator

### PR DESCRIPTION
**Pull request by sempervictus**
Create random function name generator by combining common verbs
and subjects seen in import/export tables, and sprinkling in some
random capitalization for entropy.

Testing:
  Local only via IRB -
```
>> Rex::Text.rand_func_name
=> "Releaseaddress"
>> Rex::Text.rand_func_name
=> "FreeUser"
>> Rex::Text.rand_func_name
=> "Releasefile"
>> Rex::Text.rand_func_name
=> "Describeaddress"
```

### CODE REVIEW

File: lib/rex/text/rand.rb [hunk 1]
```diff
@@ -258,5 +267,12 @@ def self.rand_mail_address
       mail_address << '@'
       mail_address << Rex::Text.rand_hostname
     end
+
+    # Generate a random funtion name
```
Comment (gwillcox-r7) [inline lib/rex/text/rand.rb:271]:
```suggestion
    # Generate a random function name
```

**Review by gwillcox-r7 — APPROVED** [review]
Looks good, just a minor typo that will need to be fixed 👍

File: lib/rex/text/rand.rb [hunk 2]
```diff
@@ -92,6 +92,15 @@ module Text
...
     ]
 
+    Func_Verb = [
+      "get", "set", "put", "describe", "acquire", "release", "initialize", "free",
+      "init", "find", "list", "create", "destroy"
+    ]
+
+    Func_Subj = [
+      "file", "directory", "dir", "address", "addr", "process", "proc", "user",
+      "priveleges", "priv", "attribute", "attr", "mutex", "handle", "type"
```
Comment (adfoster-r7) [inline lib/rex/text/rand.rb:102]:
```suggestion
      "privileges", "priv", "attribute", "attr", "mutex", "handle", "type"
```

**Comment by gwillcox-r7** [comment]
This will also need to be rebased, though I am happy to do that. The main issue I think we are looking at here is where this code would be used as right now I am struggling to find cases where one would want to use this over the already existing random text generators. Granted part of the issue is that we don't have any associated issues or code attached to this PR that we can point to as good examples of where this would be useful but perhaps you have some ideas @sempervictus? Open to ideas :)

File: lib/rex/text/rand.rb (hunk 2 above)
Comment (gwillcox-r7) [reply lib/rex/text/rand.rb]: Solved in https://github.com/rapid7/rex-text/pull/24/commits/c1634209deee80bb84b76296d8245d34cd85849e

File: lib/rex/text/rand.rb (hunk 1 above)
Comment (gwillcox-r7) [reply lib/rex/text/rand.rb]: Solved in https://github.com/rapid7/rex-text/pull/24/commits/c1634209deee80bb84b76296d8245d34cd85849e

**Comment by gwillcox-r7** [comment]
Seems to be working well, will land this and bump the Metasploit Gem version so long:

```
msf6 > irb
[*] Starting IRB shell...
[*] You are in the "framework" object

irb: warn: can't alias jobs from irb_jobs.
>> Rex::Text.rand_func_name
=> "listPriv"
>> 

```

**Pull request merged**